src/AdminService.Business/Commands/InstallAppCommand.cs

```diff
@@ -0,0 +1,166 @@
+      _repository = repository;
+      _rcCreateAdmin = rcCreateAdmin;
+      _rcCreateSmtp = rcCreateSmtp;
+      _httpContextAccessor = httpContextAccessor;
+    }
+
+    private async Task<bool> CreateSmtp(SmtpInfo smtpInfo, List<string> errors)
+    {
+      string message = "Can not create smtp credentials.";
+
+      try
+      {
+        Response<IOperationResult<bool>> response = await _rcCreateSmtp.GetResponse<IOperationResult<bool>>(
+          ICreateSmtpCredentialsRequest.CreateObj(
+            host: smtpInfo.Host,
+            port: smtpInfo.Port,
+            enableSsl: smtpInfo.EnableSsl,
+            email: smtpInfo.Email,
+            password: smtpInfo.Password));
+
+        if (response.Message.IsSuccess && response.Message.Body)
+        {
+          return true;
+        }
+
+        _logger.LogWarning(message, string.Join("\n", response.Message.Errors));
+      }
+      catch (Exception exc)
+      {
+        _logger.LogError(exc, message);
+      }
+      errors.Add(message);
+
+      return false;
+    }
+
+    private async Task<bool> CreateAdmin(AdminInfo info, List<string> errors)
+    {
+      string message = "Can not create admin.";
+
+      try
+      {
+        Response<IOperationResult<bool>> response = await _rcCreateAdmin.GetResponse<IOperationResult<bool>>(
+          ICreateAdminRequest.CreateObj(
+            info.FirstName, 
+            info.MiddleName, 
+            info.LastName, 
+            info.Email, 
+            info.Login, 
+            info.Password));
+
+        if (response.Message.IsSuccess && response.Message.Body)
+        {
+          return true;
+        }
+
+        errors.Add(message);
+
+        _logger.LogWarning(message, string.Join("\n", response.Message.Errors));
+      }
+      catch (Exception exc)
+      {
+        _logger.LogError(exc, message);
+
+        errors.Add(message);
+      }
+
+      return false;
+    }
+
+    public async Task<OperationResultResponse<Guid>> ExecuteAsync(InstallAppRequest request)
+    {
+      foreach (Guid id in request.ServicesToDisable)
+      {
+        if (await _repository.GetAsync(id) == null)
+        {
+          _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
+
+          return new OperationResultResponse<Guid>
+          {
+            Status = OperationResultStatusType.Failed,
+            Errors = new() { "Some services were not found." }
+          };
+        }
+      }
+
+      if (!_validator.ValidateCustom(request, out List<string> errors))
+      {
+        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
+
+        return new OperationResultResponse<Guid>
+        {
+          Status = OperationResultStatusType.Failed,
+          Errors = errors
+        };
+      }
+
+      if (!(await CreateSmtp(request.SmtpInfo, errors) &&
+        await CreateAdmin(request.AdminInfo, errors)))
+      {
+        return new OperationResultResponse<Guid>

```

Comment:
use ResponseCreator. 
await CreateAdmin(request.AdminInfo, errors) do you need use '!' here?

Reply:
Fixed